Indexing/CodeTree.hpp

```diff
@@ -254,65 +256,70 @@ class CodeTree
     CodeOp* _alternative;
   };
 
+  /**
+   * A search structure that collects alternatives of the same
+   * kind (either function symbols or ground terms) and offers
+   * more efficient searching and insertion over them.
+   */
   struct SearchStruct
   {
-    void destroy();
-    bool isFixedSearchStruct() const { return true; }
-
+    /**
+     * Fills out pointer @b tgt where @b insertedOp should be
+     * (or is) inserted in the structure. If @b doInsert is true
+     * an entry is inserted if not found.
+     */
+    template<bool doInsert>
     bool getTargetOpPtr(const CodeOp& insertedOp, CodeOp**& tgt);
 
+    /**
+     * Returns code op in the structure matching the content
+     * of flat term entry @b ftPos.
+     */
     CodeOp* getTargetOp(const FlatTerm::Entry* ftPos);
+    inline size_t length() const { return targets.size(); }
 
     enum Kind
     {
       FN_STRUCT,
       GROUND_TERM_STRUCT
     };
 
+    /**
+     * Actual code op for this search structure. This construction
+     * implies that search structure operations cannot be stored
+     * in @b CodeBlock containers.
+     */
     CodeOp landingOp;
     Kind kind;
+    vvector<CodeOp*> targets;
```

Comment:
Could this be `vvector<CodeOp>`?

Reply:
Doable, but upon removal we would have to shrink the vector to get rid of the non-existing entry, now we simply have null.

Reply:
Ah - OK. It could be worth doing some kind of "tombstone" logic, setting a flag in `CodeOp`? Doing the double-indirect possibly costs you a bit. But we can leave that for a future merge.